packages/flutter/test/material/text_field_test.dart

```diff
@@ -1489,4 +1489,32 @@ void main() {
       controller.selection = const TextSelection.collapsed(offset: 10);
     }, throwsFlutterError);
   });
+
+  testWidgets('Change controller to null update widget with its own controller',
+      (WidgetTester tester) async {
+    final Key key = new UniqueKey();
+    final TextEditingController controller = new TextEditingController();
+
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Material(
+          child: new TextField(
+            key: key,
+            controller: controller,
+          ),
+        ),
+      ),
+    );
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Material(
+          child: new TextField(
+            key: key,
+          ),
+        ),
+      ),
+    );
+  });
```

Comment:
we should probably have some sort of `expect` here that verifies that things are working as expected... not sure exactly what, maybe verify that the `controller` no longer does anything (but did before)?

Right now we're just verifying that it doesn't crash, which I guess it was doing before?

Reply:
>Right now we're just verifying that it doesn't crash, which I guess it was doing before?
 Write

Exactly! I didn't know what to put in expect :-/

Just added expects on controller.